### PR TITLE
Implement buildSchemaWithResolvers and lower bundle size

### PR DESCRIPTION
**Pull request by ardatan**
Related to #175
* Load all node-specific things lazily instead of adding those into the bundle.
* Remove lodash, graphql-tools etc.

### CODE REVIEW

File: src/utils/build-schema-with-resolvers.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+export interface BuildSchemaWithResolversOptions {
+    typeDefs: string | string[] | DocumentNode | DocumentNode[];
+    resolvers?: any;
+    resolversComposition?: ResolversComposerMapping;
+}
+
+export function buildSchemaWithResolvers({ typeDefs, resolvers, resolversComposition }: BuildSchemaWithResolversOptions) {
+    if (typeDefs instanceof Array) {
+        typeDefs = mergeTypeDefs(typeDefs);
```
Comment (dotansimha) [inline src/utils/build-schema-with-resolvers.ts]: I'm not sure that this method should do schema merging. maybe it should get the ready-to-use types?

File: src/utils/get-schema-directive-from-directive-resolver.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
-import { DirectiveResolverFn, SchemaDirectiveVisitor } from '@kamilkisiela/graphql-tools';
+import {  SchemaDirectiveVisitor } from '@kamilkisiela/graphql-tools/dist/schemaVisitor';
```
Comment (dotansimha) [inline src/utils/get-schema-directive-from-directive-resolver.ts]: dist?

File: src/utils/get-schema-directive-from-directive-resolver.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
 import { GraphQLField, defaultFieldResolver } from 'graphql';
+import { DirectiveResolverFn } from '@kamilkisiela/graphql-tools/dist/Interfaces';
```
Comment (dotansimha) [inline src/utils/get-schema-directive-from-directive-resolver.ts]: ?

Comment (ardatan) [reply src/utils/get-schema-directive-from-directive-resolver.ts]: Those are something I forgot related to tree-shaking.

**Comment by ardatan** [comment]
@dotansimha @dotansimha @kamilkisiela @kamilkisiela

**Pull request closed without merging**